src/werkzeug/datastructures.py

```diff
@@ -3058,7 +3058,9 @@ def save(self, dst, buffer_size=16384):
         from shutil import copyfileobj
 
         close_dst = False
-        dst = fspath(dst)
+
+        if hasattr(dst, "__fspath__"):
```

Comment:
I think this would be better written as `if isinstance(dst, os.PathLike)` in this context.

Reply:
@pgjones `hasattr(path, "__fspath__")` is backward compatible, it would work for <Py 3.6 (if there is no PathLike)

Reply:
Ah yea